From 8d08218d05a61e31c57b2eb30733da3881781968 Mon Sep 17 00:00:00 2001
From: lyintel <yang.a.lu@intel.com>
Date: Wed, 22 Jun 2022 03:48:49 +0000
Subject: [PATCH] INTERNAL: Workaround for VK_ERROR_INITIALIZATION_FAILED

Android clang add ANDROID note section, which doesn't satify 20 size.
This is a workaround for the issue.
---
 src/intel/vulkan/anv_device.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/intel/vulkan/anv_device.c b/src/intel/vulkan/anv_device.c
index 01844610dcf..46c9408ad92 100644
--- a/src/intel/vulkan/anv_device.c
+++ b/src/intel/vulkan/anv_device.c
@@ -598,12 +598,14 @@ anv_physical_device_init_uuids(struct anv_physical_device *device)
 
    unsigned build_id_len = build_id_length(note);
    if (build_id_len < 20) {
-      return vk_errorf(device, VK_ERROR_INITIALIZATION_FAILED,
-                       "build-id too short.  It needs to be a SHA");
+   /* fix me: Android clang add ANDROID note section, which does not satisfy 20 size*/
+   // return vk_errorf(device, VK_ERROR_INITIALIZATION_FAILED,
+   //                  "build-id too short.  It needs to be a SHA");
+      memcpy(device->driver_build_sha1, build_id_data(note), build_id_len);
+   } else {
+      memcpy(device->driver_build_sha1, build_id_data(note), 20);
    }
 
-   memcpy(device->driver_build_sha1, build_id_data(note), 20);
-
    struct mesa_sha1 sha1_ctx;
    uint8_t sha1[20];
    STATIC_ASSERT(VK_UUID_SIZE <= sizeof(sha1));
-- 
2.40.1

